### PR TITLE
feat: add `AssertionBuilder` to improve API check assertion usability

### PR DESCRIPTION
**Pull request by clample**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [ ] Create CLI
* [ ] Test
* [ ] Docs
* [ ] Examples
* [ ] Other

## Notes for the Reviewer

> Resolves https://github.com/checkly/checkly-cli/issues/489

Currently it's tricky to create API check assertions. We don't list the valid `source` and `comparison` values anywhere. It's also easy to create invalid assertions. For example, the `regex` field isn't needed when `source` is `STATUS_CODE`.

This PR introduces `AssertionBuilder`. This should make the possible assertion types more discoverable from IDE's autocomplete (see screenshot below). It also aims to avoid invalid assertion types from being created. 

It's still possible for users to use the old method of creating assertions. This is not a breaking change at all - just a new builder for `Assertion` objects.

#### Test Plan

I've updated the examples and made sure that `npx checkly test` and `npx checkly deploy` work. I've also checked that examples with the old assertion setup still work.

I'll also do a close review of the code to make sure that it matches our assertion checking logic and the assertion types already produced by webapp.

#### IDE Completion:

Screenshot 2023-02-09 at 16 11 31

### CODE REVIEW

**Comment by tnolet** [comment]
@clample this is already awesome. Please do not update the docs in the `README.md` as I'm moving the docs to the website. Updating examples is great.

**Comment by clample** [comment]
Thanks for checking it out @tnolet 👍 

I'll leave the docs untouched.

File: packages/cli/src/constructs/api-check.ts [hunk 1]
```diff
@@ -2,12 +2,166 @@ import { Check, CheckProps } from './check'
...
+    return this.toAssertion(AssertionComparison.NOT_NULL)
+  }
+
+  /** @private */
+  private toAssertion (comparison: AssertionComparison, target?: string|number|boolean): Assertion {
+    return {
+      source: this.source,
+      comparison,
+      property: this.property ?? '',
+      target: target !== undefined ? target.toString() : '',
```
Comment (umutuzgur) [inline packages/cli/src/constructs/api-check.ts:161]: target?.toString() ?? ''

Comment (clample) [reply packages/cli/src/constructs/api-check.ts]: Nice, that makes sense

**Review by umutuzgur — APPROVED** [review]
Looks great!

**Comment by clample** [comment]
I'll merge it now. I didn't see that it was approved.

**Pull request merged**